cirq/optimizers/align_right.py

```diff
@@ -0,0 +1,28 @@
+# Copyright 2018 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""An optimization pass that aligns gates to the right of the circuit."""
+
+from cirq import circuits, ops
+
+
+class AlignRight:
+    """Aligns gates to the right of the circuit."""
+
+    def __call__(self, circuit: circuits.Circuit):
+        self.optimize_circuit(circuit)
+
+    def optimize_circuit(self, circuit: circuits.Circuit):
+        backwards = ops.freeze_op_tree(circuit)[::-1]  # type: ignore
```

Comment:
`list(circuit.all_operations())[::-1]` should work here as well if I'm not wrong.